assets/js/stripe.js

```diff
@@ -28,11 +29,20 @@ function onSubmit(configId, form, stripe, card) {
         const serializedForm = Object.assign(JSON.parse(form.serialize(true)), {
           stripeToken: result.token.id,
           product: config.product,
-          price: parseInt(config.price.match(/\w+/g).reduce((tmp, match) => tmp + match, ''), 10),
-          price_text: config.price_text,
-          currency: config.currency,
           from: window.location.href,
         });
+
+        let price = serializedForm.price_text;
+        if (serializedForm.custom_value === "true") {
+          price = serializedForm.custom_price_text;
+          serializedForm.currency = form.$('[data-input=currency]').attr('data-value');
+        } else if (serializedForm.multichoice === "true") {
+          serializedForm.currency = form.$('input[name=price_text]:checked').attr('data-currency');
+        }
+        delete serializedForm.custom_price_text;
+        serializedForm.price_text = price;
+        serializedForm.price = parsePrice(price) * 100;
```

Comment:
Is that really always the case? Even for Japanese Yen for example?

Reply:
No not really but I'm at a loss with normalization. `parse-price` returns float if the entered value looks like a float and returns int when the value doesn't have a comma or period inside it. Also it doesn't take currency into account. It just takes out the numbers and converts comma and period into period, then casting the value to `Number`. I'm looking into other options for normalization.